oviewer/document.go

```diff
@@ -468,6 +469,27 @@ func (m *Document) WaitEOF() {
 	m.cond.Wait()
 }
 
+func (m *Document) WaitEOFWithTimeout(timeout int) {
+	if m.BufEOF() {
+		return
+	}
+
+	doneCh := make(chan struct{})
+	go func() {
+		m.cond.L.Lock()
+		defer m.cond.L.Unlock()
+		m.cond.Wait()
+		doneCh <- struct{}{}
+	}()
+
+	timeoutCh := time.After(time.Duration(timeout) * time.Millisecond)
+	select {
+	case <-doneCh:
+	case <-timeoutCh:
+		log.Println("WaitEOFWithTimeout timed out")
+	}
+}
```

Comment:
```suggestion
func (m *Document) WaitEOFWithTimeout(timeout time.Duration) {
	if m.BufEOF() {
		return
	}

	doneCh := make(chan struct{})
	go func() {
		m.cond.L.Lock()
		defer m.cond.L.Unlock()
		m.cond.Wait()
		doneCh <- struct{}{}
	}()

	select {
	case <-doneCh:
	case <-time.After(timeout):
		log.Println("WaitEOFWithTimeout timed out")
	}
}
```

Reply:
Thank you. As you say, time.Duration seems to be better.